app/src/main/java/net/squanchy/schedule/ScheduleService.java

```diff
@@ -32,53 +38,86 @@
     public Observable<Schedule> schedule() {
         Observable<FirebaseSchedule> sessionsObservable = dbService.sessions();
         Observable<FirebaseSpeakers> speakersObservable = dbService.speakers();
+        final Observable<FirebaseDays> daysObservable = dbService.days();
 
-        return Observable.combineLatest(
-                sessionsObservable,
-                speakersObservable,
-                combineIntoSchedule()
-        ).subscribeOn(Schedulers.io());
-    }
-
-    private BiFunction<FirebaseSchedule, FirebaseSpeakers, Schedule> combineIntoSchedule() {
-        return (apiSchedule, apiSpeakers) -> {
-            List<SchedulePage> pages = map(apiSchedule.days, toSchedulePage(apiSchedule, apiSpeakers));
-            return Schedule.create(pages);
-        };
+        return Observable.combineLatest(sessionsObservable, speakersObservable, combineSessionsAndSpeakers())
+                .map(mapEventsToDays())
+                .withLatestFrom(daysObservable, combineSessionsById())
+                .subscribeOn(Schedulers.io());
     }
 
-    private Lists.Function<FirebaseDay, SchedulePage> toSchedulePage(FirebaseSchedule apiSchedule, FirebaseSpeakers apiSpeakers) {
-        return apiDay -> {
-            int dayId = apiSchedule.days.indexOf(apiDay);
-            return SchedulePage.create(
-                    apiDay.date,
-                    map(apiDay.events, toEvent(apiSpeakers, dayId))
-            );
-        };
+    @NonNull
+    private BiFunction<FirebaseSchedule, FirebaseSpeakers, List<Event>> combineSessionsAndSpeakers() {
+        return (apiSchedule, apiSpeakers) -> Lists.map(apiSchedule.sessions, combineEventWith(apiSpeakers));
     }
 
-    private Lists.Function<FirebaseEvent, Event> toEvent(FirebaseSpeakers apiSpeakers, int dayId) {
+    private Lists.Function<FirebaseEvent, Event> combineEventWith(FirebaseSpeakers apiSpeakers) {
         return apiEvent -> {
             List<FirebaseSpeaker> speakers = speakersForEvent(apiEvent, apiSpeakers);
+
             return Event.create(
-                    apiEvent.eventId,
-                    dayId,      // TODO do this less crappily
+                    apiEvent.id,
+                    safelyConvertIdToInt(apiEvent.day_id),
```

Comment:
Yes, I totally missed this one, sorry. I was sure I changed it everywhere, my bad.